pylink/jlink.py

```diff
@@ -3044,11 +3044,19 @@ def register_read(self, register_index):
 
         Args:
           self (JLink): the ``JLink`` instance
-          register_index (int): the register to read
+          register_index (int/str): the register to read
 
         Returns:
           The value stored in the given register.
         """
+        # TODO: rename 'register_index' to 'register'
+        if isinstance(register_index, str):
+            regs = list(self.register_name(idx) for idx in self.register_list())
```

Comment:
I would extract this block into a classmethod (prefixed with `_`) since it's used in multiple places.  Something like:
```
@classmethod
def _register_to_name(cls, register_name):
    ...
```